src/delta_lake/data_store.py

```diff
@@ -0,0 +1,203 @@
+
+    Args:
+        database: Name of the database
+        table: Name of the table
+        spark: Optional SparkSession to use
+        return_json: Whether to return JSON string or raw data
+
+    Returns:
+        List of column names, either as JSON string or raw list
+    """
+
+    def _get_schema(session: SparkSession, db: str, tbl: str) -> List[str]:
+        try:
+            return [
+                column.name
+                for column in session.catalog.listColumns(dbName=db, tableName=tbl)
+            ]
+        except Exception as e:
+            raise DeltaSchemaError(
+                f"Error retrieving schema for table {tbl} in database {db}: {e}"
+            )
+
+    columns = _execute_with_spark(_get_schema, spark, database, table)
+    return _format_output(columns, return_json)
+
+
+def get_db_structure(
+    spark: Optional[SparkSession] = None,
+    with_schema: bool = False,
+    use_postgres: bool = True,
+    return_json: bool = True,
+) -> Union[str, Dict]:
+    """Get the structure of all databases in the Hive metastore.
+
+    Args:
+        spark: Optional SparkSession to use for operations
+        with_schema: Whether to include table schemas
+        use_postgres: Whether to use PostgreSQL for metadata retrieval
+        return_json: Whether to return the result as a JSON string
+
+    Returns:
+        Database structure as either JSON string or dictionary:
+        {
+            "database_name": ["table1", "table2"] or
+            "database_name": {
+                "table1": ["column1", "column2"],
+                "table2": ["column1", "column2"]
+            }
+        }
+    """
+
+    def _get_structure(
+        session: SparkSession,
+    ) -> Dict[str, Union[List[str], Dict[str, List[str]]]]:
+        db_structure = {}
+        databases = get_databases(spark=session, return_json=False)
+
+        for db in databases:
+            tables = get_tables(database=db, spark=session, return_json=False)
+            if with_schema:
+                db_structure[db] = {
+                    table: get_table_schema(
+                        database=db, table=table, spark=session, return_json=False
+                    )
+                    for table in tables
+                }
+            else:
+                db_structure[db] = tables
+
+        return db_structure
+
+    if use_postgres:
+        db_structure = {}
+        databases = hive_metastore.get_databases()
+
+        for db in databases:
+            tables = hive_metastore.get_tables(db)
+            if with_schema:
+                if spark is None:
+                    with get_spark_session() as spark:
+                        db_structure[db] = {
+                            table: get_table_schema(
+                                database=db, table=table, spark=spark, return_json=False
+                            )
+                            for table in tables
+                        }
```

Comment:
You could factor this out into a method to DRY things up with the lines below

Reply:
👍